Addressing instance sharing review comments

SylwekLewan-dev opened the pull request:
## Purpose
<!-- Describe the intention of the changes being proposed. What problem does it solve or functionality does it add? -->
- This PR addressed review changes from delivery of service instance sharing
- now spaces to share with are handled as a set
- naming does not have _id suffix
- uuid validation
- tests update

## Does this introduce a breaking change?
<!-- Mark one with an "x". -->

- [x] Yes
- [ ] No


## Pull Request Type

What kind of change does this Pull Request introduce?
<!-- Please check the one that applies to this PR using "X". -->

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:


## How to Test

- Test the code via automated test

```bash
go test ./...
```

<!-- Add additional steps if applicable -->
- Additional test steps

```
...
```

## What to Check

Verify that the following are valid:

- Automated tests are executed successfully
<!-- Add additional conditions if applicable -->
- ...

## Other Information
<!-- Add any other helpful information that may be needed here. -->

## Checklist for reviewer

<!-- This checklist needs to completed by the reviewer of the PR -->
The following organizational tasks must be completed before merging this PR:

- [ ] The PR is assigned to the Terraform project and a status is set (typically "in review").
- [ ] The PR has the matching labels assigned to it.
- [ ] The PR has a milestone assigned to it.
- [ ] If the PR closes an issue, the issue is referenced.
- [ ] Possible follow-up items are created and linked.

SylwekLewan-dev committed:
Addressing instance sharing review comments

JacobGsell commented on docs/resources/service_instance_sharing.md:
```diff
@@ -2,29 +2,47 @@
+  org  = data.cloudfoundry_org.my-org.id
+}
+
+# existierender service instance
```
english comment please

JacobGsell commented on docs/resources/service_instance_sharing.md:
```diff
@@ -2,29 +2,47 @@
-  service_instance_id = "e9ec29ca-993d-42e2-9c5b-cb17b1972cce"
-  space_id            = "ec6ac2b3-fb79-43c4-9734-000d4299bd59"
+  service_instance = data.cloudfoundry_service_instance.svc.id
+  spaces           = [data.cloudfoundry_space.space-to-share-with.id]
```
Please show 2 spaces to demonstrate how it will look with more than one space

JacobGsell commented on docs/resources/service_instance_sharing.md:
```diff
@@ -2,29 +2,47 @@
-- `space_id` (String) The GUID of the space to where the service instance should be shared
\ No newline at end of file
+- `service_instance` (String) The ID of the service instance to share.
+- `spaces` (Set of String) The ID of the space to share the service instance with.
```
...ID(**S**) of the space(**s**)...

JacobGsell commented on internal/provider/resource_service_instance_sharing.go:
```diff
@@ -38,22 +44,26 @@ func (r *serviceInstanceSharingResource) Schema(ctx context.Context, req resourc
 			},
-			"space_id": schema.StringAttribute{
+			"spaces": schema.SetAttribute{
 				MarkdownDescription: "The ID of the space to share the service instance with.",
```
ID(**S**) of the space(**s**)

JacobGsell commented on internal/provider/resource_service_instance_sharing.go:
```diff
@@ -84,18 +94,23 @@ func (r *serviceInstanceSharingResource) Create(ctx context.Context, req resourc
+	_, err := r.cfClient.ServiceInstances.ShareWithSpaces(ctx, plan.ServiceInstance.ValueString(), spaces)
 
+	if err != nil {
+		resp.Diagnostics.AddError("Error sharing service instance with space", err.Error())
```
... with space(**s**)

SylwekLewan-dev committed:
Sync changes from cloudfoundry main (#8)

* chore(deps): bump actions/setup-go from 5.2.0 to 5.3.0 (#111)

Bumps [actions/setup-go](https://github.com/actions/setup-go) from 5.2.0 to 5.3.0.
- [Release notes](https://github.com/actions/setup-go/releases)
- [Commits](https://github.com/actions/setup-go/compare/3041bf56c941b39c61721a86cd11f3bb1338122a...f111f3307d8850f501ac008e886eec1fd1932a34)

---
updated-dependencies:
- dependency-name: actions/setup-go
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* chore(deps): bump github.com/samber/lo from 1.47.0 to 1.49.1 (#115)

Bumps [github.com/samber/lo](https://github.com/samber/lo) from 1.47.0 to 1.49.1.
- [Release notes](https://github.com/samber/lo/releases)
- [Commits](https://github.com/samber/lo/compare/v1.47.0...v1.49.1)

---
updated-dependencies:
- dependency-name: github.com/samber/lo
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* chore(deps): bump github.com/hashicorp/terraform-plugin-go (#112)

Bumps [github.com/hashicorp/terraform-plugin-go](https://github.com/hashicorp/terraform-plugin-go) from 0.25.0 to 0.26.0.
- [Release notes](https://github.com/hashicorp/terraform-plugin-go/releases)
- [Changelog](https://github.com/hashicorp/terraform-plugin-go/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-plugin-go/compare/v0.25.0...v0.26.0)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-plugin-go
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Update for import document of space_role (#118)

* chore(deps): bump lycheeverse/lychee-action from 2.2.0 to 2.3.0 (#121)

Bumps [lycheeverse/lychee-action](https://github.com/lycheeverse/lychee-action) from 2.2.0 to 2.3.0.
- [Release notes](https://github.com/lycheeverse/lychee-action/releases)
- [Commits](https://github.com/lycheeverse/lychee-action/compare/v2.2.0...v2.3.0)

---
updated-dependencies:
- dependency-name: lycheeverse/lychee-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* chore(deps): bump golangci/golangci-lint-action from 6.2.0 to 6.3.0 (#120)

Bumps [golangci/golangci-lint-action](https://github.com/golangci/golangci-lint-action) from 6.2.0 to 6.3.0.
- [Release notes](https://github.com/golangci/golangci-lint-action/releases)
- [Commits](https://github.com/golangci/golangci-lint-action/compare/ec5d18412c0aeab7936cb16880d708ba2a64e1ae...e60da84bfae8c7920a47be973d75e15710aa8bd7)

---
updated-dependencies:
- dependency-name: golangci/golangci-lint-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Added string descriptor extension for mta (#123)

* chore(deps): bump golangci/golangci-lint-action from 6.3.0 to 6.3.2 (#125)

Bumps [golangci/golangci-lint-action](https://github.com/golangci/golangci-lint-action) from 6.3.0 to 6.3.2.
- [Release notes](https://github.com/golangci/golangci-lint-action/releases)
- [Commits](https://github.com/golangci/golangci-lint-action/compare/e60da84bfae8c7920a47be973d75e15710aa8bd7...051d91933864810ecd5e2ea2cfd98f6a5bca5347)

---
updated-dependencies:
- dependency-name: golangci/golangci-lint-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* chore: Prep Work (#129)

- Release v1.3.0

* Updated go-cfclient version (#135)

* feat: add version rule as parameter for mta resource (#130)

* Removed custom plan modifier in route resource (#137)

* chore(deps): bump github.com/hashicorp/terraform-plugin-framework (#140)

Bumps [github.com/hashicorp/terraform-plugin-framework](https://github.com/hashicorp/terraform-plugin-framework) from 1.13.0 to 1.14.0.
- [Release notes](https://github.com/hashicorp/terraform-plugin-framework/releases)
- [Changelog](https://github.com/hashicorp/terraform-plugin-framework/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-plugin-framework/compare/v1.13.0...v1.14.0)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-plugin-framework
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* chore(deps): bump github.com/hashicorp/terraform-plugin-framework-validators (#141)

Bumps [github.com/hashicorp/terraform-plugin-framework-validators](https://github.com/hashicorp/terraform-plugin-framework-validators) from 0.16.0 to 0.17.0.
- [Release notes](https://github.com/hashicorp/terraform-plugin-framework-validators/releases)
- [Changelog](https://github.com/hashicorp/terraform-plugin-framework-validators/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-plugin-framework-validators/compare/v0.16.0...v0.17.0)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-plugin-framework-validators
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Fixed removing env (#139)

* chore(deps): bump github.com/hashicorp/terraform-plugin-framework (#142)

Bumps [github.com/hashicorp/terraform-plugin-framework](https://github.com/hashicorp/terraform-plugin-framework) from 1.14.0 to 1.14.1.
- [Release notes](https://github.com/hashicorp/terraform-plugin-framework/releases)
- [Changelog](https://github.com/hashicorp/terraform-plugin-framework/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-plugin-framework/compare/v1.14.0...v1.14.1)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-plugin-framework
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* feat: add service plan data source

* feat: Add cf-cli (#151)

Install cf-cli during the build of the devcontainer

* test: update OpenTofu test matrix

* Added test cases for service data plan

* chore(deps): bump actions/setup-go from 5.3.0 to 5.4.0

Bumps [actions/setup-go](https://github.com/actions/setup-go) from 5.3.0 to 5.4.0.
- [Release notes](https://github.com/actions/setup-go/releases)
- [Commits](https://github.com/actions/setup-go/compare/f111f3307d8850f501ac008e886eec1fd1932a34...0aaccfd150d50ccaeb58ebd88d36e91967a5f35b)

---
updated-dependencies:
- dependency-name: actions/setup-go
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* feat: added support for service plan visibility  (#161)

* feat: added support for service plan visibility

* feat: update service plan visibility

* chore: update service plan visibility

* feat: update plan visibility

* feat: update service plan visibility

* fix: fixes the failing test for service plan visibility

* fix: update space attribute to be computed

* feat: update markdown description for delete function

* fix: update markdown description for delete function

* fix: markdown decsription update

* update markdown description

* Added cf only user resource

* chore(deps): bump github.com/hashicorp/terraform-plugin-testing

Bumps [github.com/hashicorp/terraform-plugin-testing](https://github.com/hashicorp/terraform-plugin-testing) from 1.11.0 to 1.12.0.
- [Release notes](https://github.com/hashicorp/terraform-plugin-testing/releases)
- [Changelog](https://github.com/hashicorp/terraform-plugin-testing/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-plugin-testing/compare/v1.11.0...v1.12.0)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-plugin-testing
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* test: update Terraform test matrix

* Rerecorded provider tests for v1.10.* above

* chore(deps): bump golangci/golangci-lint-action from 6.3.2 to 6.5.2

Bumps [golangci/golangci-lint-action](https://github.com/golangci/golangci-lint-action) from 6.3.2 to 6.5.2.
- [Release notes](https://github.com/golangci/golangci-lint-action/releases)
- [Commits](https://github.com/golangci/golangci-lint-action/compare/v6.3.2...55c2c1448f86e01eaae002a5a3a9624417608d84)

---
updated-dependencies:
- dependency-name: golangci/golangci-lint-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* chore(deps): bump github.com/cloudfoundry-community/go-uaa

Bumps [github.com/cloudfoundry-community/go-uaa](https://github.com/cloudfoundry-community/go-uaa) from 0.3.4-0.20240727153833-d675ee37e6c9 to 0.3.5.
- [Release notes](https://github.com/cloudfoundry-community/go-uaa/releases)
- [Commits](https://github.com/cloudfoundry-community/go-uaa/commits/v0.3.5)

---
updated-dependencies:
- dependency-name: github.com/cloudfoundry-community/go-uaa
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

* chore: v1.4.0 Prep Work

- Release v1.4.0

* Added comment for user_cf import

* chore: Add note

- Add note to user data source

* feat: network_policy resource added (#171)

* Add NetworkPolicy resource

* Generate documentation

* Add a test for updating network policy resource

* Remove unused networkPoliciesType.Id

* Move policy_client from session to resource_network_policy

* Add network policy resource to provider test

* Regenerate docs

* chore(deps): bump lycheeverse/lychee-action from 2.3.0 to 2.4.0 (#174)

Bumps [lycheeverse/lychee-action](https://github.com/lycheeverse/lychee-action) from 2.3.0 to 2.4.0.
- [Release notes](https://github.com/lycheeverse/lychee-action/releases)
- [Commits](https://github.com/lycheeverse/lychee-action/compare/v2.3.0...v2.4.0)

---
updated-dependencies:
- dependency-name: lycheeverse/lychee-action
  dependency-version: 2.4.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Lint action update  (#176)

* chore(deps): bump golangci/golangci-lint-action from 6.5.2 to 7.0.0

* Migrated the lint configuration from v1 to v2

* Fixed the static check errors coming after updating the linter version

* fix:route destination issue with app

* fix : inconsistent plan for private docker app image

* feat: add module as parameter for mta resource

Allows specifying modules that should be deployed.

* feat:Add container ssh control on app level

* Added deploy logs in TF_LOG

* fix : make spaces in security group computed

* chore : v1.5.0 Prep Work

* chore(deps): bump github.com/samber/lo from 1.49.1 to 1.50.0

Bumps [github.com/samber/lo](https://github.com/samber/lo) from 1.49.1 to 1.50.0.
- [Release notes](https://github.com/samber/lo/releases)
- [Commits](https://github.com/samber/lo/compare/v1.49.1...v1.50.0)

---
updated-dependencies:
- dependency-name: github.com/samber/lo
  dependency-version: 1.50.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* chore(deps): bump code.cloudfoundry.org/lager/v3 from 3.30.0 to 3.36.0

Bumps [code.cloudfoundry.org/lager/v3](https://github.com/cloudfoundry/lager) from 3.30.0 to 3.36.0.
- [Release notes](https://github.com/cloudfoundry/lager/releases)
- [Commits](https://github.com/cloudfoundry/lager/compare/v3.30.0...v3.36.0)

---
updated-dependencies:
- dependency-name: code.cloudfoundry.org/lager/v3
  dependency-version: 3.36.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* chore(deps): bump code.cloudfoundry.org/policy_client

Bumps [code.cloudfoundry.org/policy_client](https://github.com/cloudfoundry/policy_client) from 0.47.0 to 0.52.0.
- [Release notes](https://github.com/cloudfoundry/policy_client/releases)
- [Commits](https://github.com/cloudfoundry/policy_client/compare/v0.47.0...v0.52.0)

---
updated-dependencies:
- dependency-name: code.cloudfoundry.org/policy_client
  dependency-version: 0.52.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* fix : params instance binding in app

* chore(deps): bump code.cloudfoundry.org/policy_client

Bumps [code.cloudfoundry.org/policy_client](https://github.com/cloudfoundry/policy_client) from 0.52.0 to 0.53.0.
- [Release notes](https://github.com/cloudfoundry/policy_client/releases)
- [Commits](https://github.com/cloudfoundry/policy_client/compare/v0.52.0...v0.53.0)

---
updated-dependencies:
- dependency-name: code.cloudfoundry.org/policy_client
  dependency-version: 0.53.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* chore(deps): bump golangci/golangci-lint-action from 7.0.0 to 8.0.0

Bumps [golangci/golangci-lint-action](https://github.com/golangci/golangci-lint-action) from 7.0.0 to 8.0.0.
- [Release notes](https://github.com/golangci/golangci-lint-action/releases)
- [Commits](https://github.com/golangci/golangci-lint-action/compare/1481404843c368bc19ca9406f87d6e0fc97bdcfd...4afd733a84b1f43292c63897423277bb7f4313a9)

---
updated-dependencies:
- dependency-name: golangci/golangci-lint-action
  dependency-version: 8.0.0
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

* chore(deps): bump code.cloudfoundry.org/lager/v3 from 3.36.0 to 3.37.0

Bumps [code.cloudfoundry.org/lager/v3](https://github.com/cloudfoundry/lager) from 3.36.0 to 3.37.0.
- [Release notes](https://github.com/cloudfoundry/lager/releases)
- [Commits](https://github.com/cloudfoundry/lager/compare/v3.36.0...v3.37.0)

---
updated-dependencies:
- dependency-name: code.cloudfoundry.org/lager/v3
  dependency-version: 3.37.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* chore(deps): bump actions/setup-go from 5.4.0 to 5.5.0

Bumps [actions/setup-go](https://github.com/actions/setup-go) from 5.4.0 to 5.5.0.
- [Release notes](https://github.com/actions/setup-go/releases)
- [Commits](https://github.com/actions/setup-go/compare/0aaccfd150d50ccaeb58ebd88d36e91967a5f35b...d35c59abb061a4a6fb18e82ac0862c26744d6ab5)

---
updated-dependencies:
- dependency-name: actions/setup-go
  dependency-version: 5.5.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* chore(deps): bump lycheeverse/lychee-action from 2.4.0 to 2.4.1

Bumps [lycheeverse/lychee-action](https://github.com/lycheeverse/lychee-action) from 2.4.0 to 2.4.1.
- [Release notes](https://github.com/lycheeverse/lychee-action/releases)
- [Commits](https://github.com/lycheeverse/lychee-action/compare/v2.4.0...v2.4.1)

---
updated-dependencies:
- dependency-name: lycheeverse/lychee-action
  dependency-version: 2.4.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

* chore(deps): bump code.cloudfoundry.org/policy_client

Bumps [code.cloudfoundry.org/policy_client](https://github.com/cloudfoundry/policy_client) from 0.53.0 to 0.54.0.
- [Release notes](https://github.com/cloudfoundry/policy_client/releases)
- [Commits](https://github.com/cloudfoundry/policy_client/compare/v0.53.0...v0.54.0)

---
updated-dependencies:
- dependency-name: code.cloudfoundry.org/policy_client
  dependency-version: 0.54.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* chore(deps): bump github.com/hashicorp/terraform-plugin-framework-validators

Bumps [github.com/hashicorp/terraform-plugin-framework-validators](https://github.com/hashicorp/terraform-plugin-framework-validators) from 0.17.0 to 0.18.0.
- [Release notes](https://github.com/hashicorp/terraform-plugin-framework-validators/releases)
- [Changelog](https://github.com/hashicorp/terraform-plugin-framework-validators/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-plugin-framework-validators/compare/v0.17.0...v0.18.0)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-plugin-framework-validators
  dependency-version: 0.18.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Sarthak Goyal <61136876+sp1goyal@users.noreply.github.com>
Co-authored-by: Debaditya Ray <65864761+Dray56@users.noreply.github.com>
Co-authored-by: Vipin Menon <vipinvkmenon@users.noreply.github.com>
Co-authored-by: Philipp <philipp@stotz.email>
Co-authored-by: Christian Lechner <22294087+lechnerc77@users.noreply.github.com>
Co-authored-by: Debaditya <debadityaray00@gmail.com>
Co-authored-by: PRAJIN OLLEKKATT PRAKASAN <71236507+prajin-op@users.noreply.github.com>
Co-authored-by: Vipin Vijaykumar <vipin.vijaykumar@sap.com>
Co-authored-by: Ryan Ahearn <ryan.ahearn@gsa.gov>
Co-authored-by: Philipp Stotz <philipp.stotz@sap.com>
